textx/metamodel.py

```diff
@@ -561,23 +562,31 @@ def model_from_str(self, model_str, file_name=None, debug=None,
         if file_name is None:
             model = self._parser_blueprint.clone().get_model_from_str(
                 model_str, debug=debug,
-                pre_ref_resolution_callback=pre_ref_resolution_callback)
+                pre_ref_resolution_callback=pre_ref_resolution_callback,
+                project_root=project_root)
 
             for p in self._model_processors:
                 p(model, self)
         else:
-            model = self.internal_model_from_file(file_name, encoding, debug,
-                                                  model_str=model_str)
+            model = self.internal_model_from_file(file_name,
+                                                  encoding=encoding,
+                                                  debug=debug,
+                                                  model_str=model_str,
+                                                  project_root=project_root)
 
```

Comment:
Good! The new "API" with model parameters (#240) should be compatible to your proposal!